axum-extra/src/extract/cookie/signed.rs

```diff
@@ -63,22 +61,36 @@ use std::{convert::Infallible, fmt, marker::PhantomData};
 ///     # todo!()
 /// }
 ///
-/// // Generate a secure key
-/// //
-/// // You probably don't wanna generate a new one each time the app starts though
-/// let key = Key::generate();
+/// // our application state
+/// #[derive(Clone)]
+/// struct AppState {
+///     // that holds the key used to sign cookies
+///     key: Key,
+/// }
+///
+/// // this impl tells `SignedCookieJar` how to access the key from our state
+/// impl FromRef<AppState> for Key {
+///     fn from_ref(state: &AppState) -> Self {
+///         state.key.clone()
+///     }
+/// }
```

Comment:
Is it a good idea to recommend implementing `FromRef` for a cloned version that requires `.clone()` instead of a borrowed version? The `FromRef` implementations from the docs seemed to always have `.clone()`.

Reply:
How else would you do it?

Reply:
Maybe recommend the borrowed version like `impl FromRef<AppState> for &Key` in cases where clone is heavy and user does not need the owned version? Like show an example.

Reply:
Yeah exactly. I don't think it'll actually work.